datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -232,6 +241,74 @@ impl ParquetExec {
         self.enable_page_index
             .unwrap_or(config_options.execution.parquet.enable_page_index)
     }
+
+    /// Redistribute files across partitions according to their size
+    pub fn get_repartitioned(&self, target_partitions: usize) -> Self {
+        // Perform redistribution only in case all files should be read from beginning to end
+        let has_ranges = self
+            .base_config()
+            .file_groups
+            .iter()
+            .flatten()
+            .any(|f| f.range.is_some());
+        if has_ranges {
+            return self.clone();
+        }
+
+        let total_size = self
+            .base_config()
+            .file_groups
+            .iter()
+            .flatten()
+            .map(|f| f.object_meta.size as i64)
+            .sum::<i64>();
+        let target_partition_size =
```

Comment:
I think we should probably have a minimum file range, anything less than 10MB is unlikely to yield additional parallelism, as row groups shouldn't be this small, and adding more partitions is not necessarily free.

We could make this configurable

Reply:
Sounds reasonable, but it seems to me that same thing can be done via both `target_partitions` & `parallel_file_scan` if we know file layout prior to running query - is this setting supposed to be kind of fuse for running queries over random files (i.e. neither the number of files nor their size is known)?

Reply:
After some thoughts, I see that it could be just more user-friendly -- cause this setting will care about preventing unnecessary file splits without any additional settings tweaking efforts.

I've also moved both of these new settings from `execution` to `optimizer` section.